### ISSUE TITLE
Move password-manager to a separate repository

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#637]
I'd like to move the password manager to a separate repository and publish it on Quicklisp.
Maybe it'd need a slight overhaul though, considering it's rather trivial and maybe not general enough.  Thoughts?

**Comment by jmercouris** [comment]
I think it needs some generalization as well, a small amount of polishing. I would first move it, and then polish it and generalize it. I personally want to add support for the macOS keychain, as well as other built in OS keychains like kde wallet etc

**Comment by Ambrevar** [comment]
In particular, we should make sure it's extensible.  Right now it's impossible for users to extend it to support a new interface because of how `make` hardcode the list of supported interfaces.

**Comment by jmercouris** [comment]
I agree with that, it was even difficult for me to extend. The API needs to be changed a little bit I think (to accomodate different tools). How should the API be changed? I'm not sure.

**Comment by Ambrevar** [comment]
I've updated the `make` function to initialize the password from the `interface-list` variable which is set by the various interfaces and can be set by the user.

**Comment by jmercouris** [comment]
Great! Thank you!

**Comment by aadcg** [comment]
I think it was premature to start hosting Nyxt libraries as separate repositories as it increases the maintenance. Proper library demarcation is an healthy practice, as to re-use code.

Let's shift gears and accumulate libraries under `libraries/`.
